dataScience/src/text_classif/README.md

```diff
@@ -0,0 +1,224 @@
+columns in the input `.csv`. 
+
+See the documentation in `predict_cli.py` for usage. 
+
+## Test Data
+The benchmark test data is the *Corpus of Linguistic Acceptability* (CoLA) and is included in the `tests/test_data/cola_public`
+directory. The full data set has 8,551 sentences labeled as 0 or 1 (*not grammatical*, *grammatical*).
+See [Warstadt *et al*., (2019)](https://arxiv.org/pdf/1805.12471.pdf) for additional detail.
+
+The directory `test_data/responsibility` has Gamechanger-specific data that can also be
+used for testing purposes.
+
+## Training
+`example_cola_cli.py` will train a classification model on the CoLA data.
+
+For `CONFIG_YAML`, the file `sample_distilbert_cola_config.yml` has the necessary configuration. The
+`DATA_FILE` should point to `/tests/test_data/cola_public/raw/in_domain_train.tsv`. Finally, `-m distilbert`
+will train a model using `distilbert-base-uncased`.
+
+The sample configuration file should be updated if you wish to save a checkpoint or change the
+model parameters such as the number of epochs.
+
+
+## Logging and Metrics
+Console and/or file logging is supported via `utills/initialize_logger()`. 
+See `examples/example_gc_cli.py` for example usage. You'll need to alter the
+configuration file to accommodate your paths and names.
+
+At the end of each epoch, a `scikit-learn` classification report is logged as well
+as a formatted confusion matrix.
+
+## Subclassing
+Since different models
+require their own `transformers` class for tokenizing and sequence classification,
+subclassing `Classifier` allows different models to be trained and tested. 
+
+A subclass must implement the following method:
+
+- `load_model_tokenizer()`
+
+The modules `roberta_classifier.py`, `bert_classifier.py`, and `distilbert_classifier.py` are examples of subclasses. These
+can be further subclassed for your custom methods.
+
+## Example Metrics
+```
+[classifier.py:511 - _validate()], 
+
+              precision    recall  f1-score   support
+
+           0      0.789     0.418     0.546       268
+           1      0.782     0.949     0.857       588
+
+    accuracy                          0.783       856
+   macro avg      0.785     0.683     0.702       856
+weighted avg      0.784     0.783     0.760       856
+
+[classifier.py:512 - _validate()], confusion matrix
+
+	         pred: 0  pred: 1
+true: 0      112      156
+true: 1       30      558
+
+[classifier.py:513 - _validate()], 	validation loss : 0.534
+[classifier.py:514 - _validate()], 	            MCC : 0.457
+[classifier.py:515 - _validate()], 	            AUC : 0.683
+[classifier.py:516 - _validate()], 	 accuracy score : 0.783
+[classifier.py:517 - _validate()], 	validation time : 0:01:48
+[checkpoint_handler.py:30 - write_checkpoint()], saving model with loss : 0.534
+```
+
+## License
+The MIT License (MIT)
+
+Copyright &#169; 2021 Chris Skiscim, Amina Jackson; Booz Allen Hamilton, Inc.
```

Comment:
We are not copyrighting this code this way. The code is licensed under MIT with no BAH copyrights.

Reply:
license is in `<repo>/LICENSE.md` so this is redundant anyway